packages/form-layout/src/layouts/auto-responsive-layout-controller.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2025 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { isElementHidden } from '@vaadin/a11y-base/src/focus-utils';
+import { AbstractLayoutController } from './abstract-layout-controller.js';
+
+/**
+ * Check if the node is a line break element.
+ *
+ * @param {HTMLElement} el
+ * @return {boolean}
+ */
+function isBreakLine(el) {
+  return el.localName === 'br';
+}
+
+/**
+ * A controller that implements the auto-responsive layout algorithm.
+ * Not intended for public use.
+ *
+ * @private
+ */
+export class AutoResponsiveLayoutController extends AbstractLayoutController {
+  constructor(host) {
+    super(host, {
+      mutationObserverOptions: {
+        subtree: true,
+        childList: true,
+        attributes: true,
+        attributeFilter: ['colspan', 'data-colspan', 'hidden'],
+      },
+    });
+  }
+
+  /** @override */
+  connect() {
+    if (this.isConnected) {
+      return;
+    }
+
+    super.connect();
+
+    this.updateLayout();
+  }
+
+  /** @override */
+  disconnect() {
```

Comment:
Reactive controllers actually have `hostConnected()` and `hostDisconnected()` called on the respective host callbacks and it's also supported by `ControllerMixin` - would it make sense to use these?

Reply:
Renamed the classes to AbstractLayout, AutoResponsiveLayout, etc.